src/services/environment-detection-service.ts

```
@@ -0,0 +1,188 @@
+    timeout: number;
+  };
+}
+
+export class EnvironmentDetectionService {
+  private static instance: EnvironmentDetectionService;
+  private config: EnvironmentConfig | null = null;
+
+  private constructor() {}
+
+  static getInstance(): EnvironmentDetectionService {
+    if (!this.instance) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)